File: angular/src/app/domain/poi.ts
import {EntityType} from '@shared/domain/entities';

/**
 * Point of Interest
 *   {
 *   "id": "d5685229-2690-44ef-8772-fb0616dfe2d9",
 *   "name": "Fish Mekong Delta",
 *   "imageUrl": "https://img.youtube.com/vi/134.jpg",
 *   "entityType": "Post",
 *   "coordinates": [
 *     105.746854,
 *     10.045162
 *   ]
 * },
 */
export interface POI {
  id?: string;
  name: string;
  areaCode?: string;
  imageUrl?: string;
  coordinates?: Array<number>;
  locationType?: string;
  entityType?: EntityType;
}

File: kotlin/src/main/kotlin/net/timafe/angkor/domain/dto/POI.kt
package net.timafe.angkor.domain.dto

import org.springframework.beans.factory.annotation.Value
import java.util.*

/**
 * Lightweight Point Of Interest representation,
 * basically coordinates with a name and area attached
 */
interface POI {
    // ("id","name","areaCode","imageUrl","locationType","coordinates")
    var id: UUID
    var name: String
    var areaCode: String
    var imageUrl: String?
    var locationType: String?

    // coordinates should be List<Double>? but this didn't work with JPA SELECT NEW query
    // (see PlaceRepository) which raises
    // Expected arguments are: java.util.UUID, java.lang.String, java.lang.Object
    // var coordinates: java.lang.Object? = null
    // override var coordinates: List<Double> = listOf()
    @Value("#{@mappingService.postgresCoordinateStringToList(target.coordinates)}")
    fun getCoordinates(): List<Double>

}
